gcc/rust/hir/tree/rust-hir.h

```diff
@@ -45,6 +45,41 @@ class HIRTypeVisitor;
 // forward decl for use in token tree method
 class Token;
 
+// Kind for downcasting various HIR nodes to other base classes when visiting
+// them
+enum BaseKind
```

Comment:
i like this alot!

Reply:
Usually we put this under the Node class but its up to your wether you want to do that or not it might be annoying

Reply:
Done in 1df917c0723